walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +273,141 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
-}
 
+  @Test
+  public void optionalAttendeeBusy() {
+    // 1. Have each person have different events. Add an optional attendee C who has an all-day event. The same three time slots should be returned as when C was not invited.
+    //
+    // Events  :       |--A--|     |--B--|
+    //           |---------C-optional----------| 
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.WHOLE_DAY,
+        Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B ), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optional830_9() {
+    // 2. Have each person have different events. Add an optional attendee C who has an all-day event. The same three time slots should be returned as when C was not invited.
```

Comment:
This comment doesn't match the diagram below. I also think in this case only times 1 and 3 should be returned, since there are time slots that accomodate C.